### ISSUE TITLE
Air Quality Index

### ISSUE THREAD

**Issue by queuebit** [Pirate-Weather/pirateweather#92]
I live in the Midwestern part of the United States and we are seeing pollutants from forest fires in Canada often, recently.

Is there a way to pull some standard Air Quality value into the current or forecast data?

Thanks!

**Comment by DaveL17** [comment]
I live in the midwest as well. I've been pulling air quality data from [Open Weather](https://api.openweathermap.org/data/2.5/air_pollution).

**Comment by alexander0042** [comment]
Great idea, and something I'm looking into! I'm based out of Ottawa, so lots of smoke here too. The trick for this will be finding a source, since I don't want to just pull from another API, as that kind of defeats the point of this project. 

My top pick is the [HRRR Smoke model](https://rapidrefresh.noaa.gov/hrrr/HRRRsmoke/). It only covers southern Canada/ US, but it would be a starting point.

**Comment by dustinbolton** [comment]
@alexander0042 You can get raw air quality data sources for both observation and forecasts for the USA at https://www.airnowapi.org (data files directly at https://files.airnowtech.org/ ) for free. Documentation and API access comes with the free account.

**Comment by alexander0042** [comment]
This remains a great idea, and something that I want to look into; however, since it's not part of any of the models that I'm currently ingesting, I think it's out of scope for V2.0. I'm going to put this link here: <https://www.nco.ncep.noaa.gov/pmb/products/aqm/>, since I think this is the best option for adding this in in the future. That being said,

**Comment by cloneofghosts** [comment]
Does the model you posted have global coverage or is it for CONUS only? If that model is for CONUS only you'd probably would still need to look for one with global coverage.

The model I see most sites using is CAMS which I posted the link to above but will also post again: https://ads.atmosphere.copernicus.eu/cdsapp#!/dataset/cams-europe-air-quality-forecasts?tab=overview and https://ads.atmosphere.copernicus.eu/cdsapp#!/dataset/cams-global-atmospheric-composition-forecasts?tab=overview

Someone else posted a link to Airnow which has grib files for current Ozone, PM10 and PM2.5 based on current observation data. Not sure of the coverage area but I think it would be the US and parts of Canada? https://www.airnowapi.org/ (data files directly at https://files.airnowtech.org/ )

**Comment by Tarun80** [comment]
With Pirate Weather being based on Dark Sky, which Apple was using. Could also base a few other things on what they use in their Weather app.

For Air Quality (and some maps as well) they use several things, but here's [the data sources](https://support.apple.com/en-ca/105038).

For Air Quality, they refer to the [Breezometer](https://www.breezometer.com/air-quality-map/) - which even having the maps in Pirate Weather would be awesome.


Hope this helps!

**Comment by cloneofghosts** [comment]
@alexander0042 Is this something you're still thinking about adding? It's not listed in #414 so am unsure when this might be addressed.

**Comment by DaveL17** [comment]
Bump to avoid automatic closure.

**Comment by alexander0042** [comment]
Was just updating the roadmap and remembered this. The API is currently incorporating data on particulate (<2.5um) and ozone information, so I'm wondering if that accomplishes what we set out to do here, or if there's another source that would be worth pulling in.

**Comment by DaveL17** [comment]
I see `ozone` as currently and hourly, but I couldn't spot `particulate` in the API docs. Did I just flat miss it?

**Comment by alexander0042** [comment]
Good question! The `smoke` variable shows the particulate matter- it's not limited to just smoke based particulates (although I should check this)

**Comment by DaveL17** [comment]
Yep -- it's there. I live in the Midwest US and the only time we think about smoke is when it belongs to someone else. I think this issue can be closed as implemented. Thanks!

**Comment by cloneofghosts** [comment]
> Was just updating the roadmap and remembered this. The API is currently incorporating data on particulate (<2.5um) and ozone information, so I'm wondering if that accomplishes what we set out to do here, or if there's another source that would be worth pulling in.

The `smoke` parameter only shows data for areas in the CONUS region and elsewhere is just -999 plus particulate (<2.5um) isn't necessarily just smoke. Is the `ozone` parameter in the API similar to what other sources report for ozone or is it different? Also the Air Quality index is normally calculated with more parameters: pm10, no2, co and so2 which the API doesn't have currently.

**Comment by cloneofghosts** [comment]
@alexander0042 I think this should have Air Quality data https://registry.opendata.aws/silam/. When I try to browse the buckets nothing appears but their site does have data https://silam.fmi.fi/index.html

They also have smoke forecasts as well which might be a good addition for a global smoke forecast.

**Comment by alexander0042** [comment]
I really like FMI as a source, since I suspect it's pretty good data! Also already gridded and NetCDF data, so pretty ideal. Probably the sort of thing we hide behind an include, but I think worth doing. Thanks for finding the source!

**Comment by KnownStormChaser** [comment]
I have some other air quality model sources if you're looking for more options.

CAMS is probably the most widely used, they have a global and EU model:
https://ads.atmosphere.copernicus.eu/datasets/cams-global-atmospheric-composition-forecasts?tab=overview
https://ads.atmosphere.copernicus.eu/datasets/cams-europe-air-quality-forecasts?tab=overview

NOAA Air Quality Model (AQM) is best for CONUS, Alaska and Hawaii:
https://registry.opendata.aws/noaa-nws-naqfc-pds/

The Canadian RAQDPS covers most of North America and is great for forest fire smoke:
https://eccc-msc.github.io/open-data/msc-data/nwp_raqdps/readme_raqdps_en/

Germany has ART versions of their ICON global and EU models. (Global is for Saharan dust and EU is for pollen):
https://opendata.dwd.de/weather/nwp/v1/m/icon-art/p/
https://opendata.dwd.de/weather/nwp/v1/m/icon-art-eu/p/

**Comment by ghosttie** [comment]
So does this just feed into the existing properties in the API or are there new ones?

**Comment by cloneofghosts** [comment]
> So does this just feed into the existing properties in the API or are there new ones?

There are new ones. Added fields are

airQualityIndex
coConcentration
no2Concentration
ozoneConcentration
so2Concentration
pm10
pm25

airQualityIndex is behind the version=2 flag and the others behind a include=airqualitydetails flag. It's being pushed up to dev endpoint at the moment so you should be able to test before it goes live.

Should also note that airQualityIndex is dependant on the units.

ca -> ECCC AQHI
us/default -> US EPA
si/us -> EU CAQI

**Comment by alexander0042** [comment]
Sorry for taking so long to get it- because we had some existing sources, there were other pieces that I wanted to get in place first. I really like the RAQDPS/SILAM models though, and hopefully the addition of their data will answer what you're looking for here!
